calliope/test/test_backend_pyomo.py

```diff
@@ -635,6 +639,31 @@ def test_loc_techs_cost_var_constraint(self, tech, scenario, cost):
         m.run(build_only=True)
         assert hasattr(m._backend_model, "cost_var_constraint")
 
+    def test_one_way_om_cost(self):
+        """
+        With one_way transmission, it should still be possible to set an om_prod cost.
+        """
+        m = build_model(
+            {
+                "techs.test_transmission_elec.costs.monetary.om_prod": 1,
+                "links.0,1.techs.test_transmission_elec.constraints.one_way": True,
+            },
+            "simple_supply,two_hours",
+        )
+        m.run(build_only=True)
+        has_cost = get_indexed_constraint_body(
+            m._backend_model,
+            "cost_var_constraint",
+            ("monetary", "1::test_transmission_elec:0", m._backend_model.timesteps[1]),
+        ).to_string()
+        has_no_cost = get_indexed_constraint_body(
+            m._backend_model,
+            "cost_var_constraint",
+            ("monetary", "0::test_transmission_elec:1", m._backend_model.timesteps[1]),
+        ).to_string()
```

Comment:
I think it would be good if you could build the arguments to `get_indexed_constraint_body` in one variable, and vary it to get the second one.  In the current way, you have to look very carefully to spot where the differences are.  So something like this maybe:

```suggestion
        arg1, arg2 = m._backend_model, "cost_var_constraint",
        arg3 = ["monetary", "1::test_transmission_elec:0", m._backend_model.timesteps[1]]
        has_cost = get_indexed_constraint_body(arg1, arg2, arg3).to_string()
        
        arg3[1] =  "0::test_transmission_elec:1"
        has_no_cost = get_indexed_constraint_body(arg1, arg2, arg3).to_string()
```
This way it's very explicit.  If the third argument needs to be a tuple, you can always pass `tuple(arg3)`.